src/python/pants/backend/shell/target_types.py

```diff
@@ -442,6 +461,41 @@ class ShellCommandTarget(Target):
     )
 
 
+class ShellRunInSandboxTarget(Target):
+    alias = "experimental_run_in_sandbox"
+    core_fields = (
+        *COMMON_TARGET_FIELDS,
+        RunInSandboxRunnableField,
+        RunInSandboxArgumentsField,
+        ShellCommandExecutionDependenciesField,
+        ShellCommandOutputDependenciesField,
+        ShellCommandLogOutputField,
+        ShellCommandOutputsField,
+        RunInSandboxSourcesField,
+        ShellCommandTimeoutField,
+        ShellCommandToolsField,
+        ShellCommandExtraEnvVarsField,
+        EnvironmentField,
+    )
+    help = softwrap(
+        """
+        Execute any runnable target for its side effects.
+
+        Example BUILD file:
+
+            experimental_run_in_sandbox(
+                runnable=":python_source",
+                argv=[""],
+                tools=["tar", "curl", "cat", "bash", "env"],
+                data_dependencies=[":scripts"],
```

Comment:
Two small questions:

- Is this meant to be demonstrating that `:python_source` can internally run this scripts? 
- Is this field name equivalent to the new `execution_dependencies` for `esc`? (or maybe `output_dependencies`?)

Reply:
I think this example definition was pulled from a code generation example, that runs a Python program that consume other Python sources as data (e.g. a web backend with API annotations). With that in mind, the docstring says that `:python_source` should be run, and `:scripts` should be made available as dependencies.

I think `data_dependencies` was meant to say `execution_dependencies`, and may be a holdover from an earlier iteration. I'll fix that, along with the rest of the docstring there.

* It's saying that the target defined by `:python_source` can be run
Yes, that docstring should indeed say `data_dependencies`, and